File: fibonacci/fibo2.py
import time

def fibonacci(n):
    value = 0
    f1 = 1
    f2 = -1
    for i in range(n+1):
        value = f1 + f2
        f2 = f1
        f1 = value
    
    return value

if __name__ == "__main__":
    for x in range(3):
        fibonacci(3)
    time.sleep(0)
    start = time.time()
    for x in range(0, 5000):
        fibonacci(x)
    end = time.time()
    print end - start

File: fibonacci/fibo2.java
import java.math.BigInteger;

public class fibo2 {
    public static BigInteger fibonacci(int n) {
        BigInteger value = new BigInteger("0");
        BigInteger f1 = new BigInteger("1");
        BigInteger f2 = new BigInteger("-1");
        for (int i=0; i< n+1; i++) {
            value = f1.add(f2);
            f2 = f1;
            f1 = value;
        }

        return value;
    }
    public static void main(String[] argv) {
        long start = System.currentTimeMillis();
        for (int i=0; i<5000; i++) {
            fibonacci(i);
        }
        long end = System.currentTimeMillis();
        System.out.println(1.0*(end - start)/1000);
    }
}
